tests/test_project.py

```diff
@@ -1775,6 +1776,178 @@ class TestGroupDynamicProjectMainInterface(TestProjectMainInterface):
     project_class = _DynamicTestProject
 
 
+class TestAggregatesProjectBase(TestProjectBase):
+    project_class = _TestAggregateProject
+    entrypoint = dict(
+        path=os.path.realpath(
+            os.path.join(os.path.dirname(__file__), "define_test_aggregate_project.py")
+        )
+    )
+
+    def mock_project(self):
+        project = self.project_class.get_project(root=self._tmp_dir.name)
+        for i in range(1, 31):
+            project.open_job(dict(i=i, even=bool(i % 2 == 0))).init()
+        project = project.get_project(root=self._tmp_dir.name)
+        project._entrypoint = self.entrypoint
+        return project
+
+    def switch_to_cwd(self):
+        os.chdir(self.cwd)
+
+    @pytest.fixture(autouse=True)
+    def setup_main_interface(self, request):
+        self.project = self.mock_project()
+        self.cwd = os.getcwd()
+        os.chdir(self._tmp_dir.name)
+        request.addfinalizer(self.switch_to_cwd)
+
+    def call_subcmd(self, subcmd):
+        # Determine path to project module and construct command.
+        fn_script = inspect.getsourcefile(type(self.project))
+        _cmd = f"python {fn_script} {subcmd}"
+        try:
+            with add_path_to_environment_pythonpath(os.path.abspath(self.cwd)):
+                with switch_to_directory(self.project.root_directory()):
+                    return subprocess.check_output(
+                        _cmd.split(), stderr=subprocess.DEVNULL
+                    )
+        except subprocess.CalledProcessError as error:
+            print(error, file=sys.stderr)
+            print(error.output, file=sys.stderr)
+            raise
+
+
+class TestProjectUtilities(TestAggregatesProjectBase):
+    def test_AggregatesCursor(self):
+        project = self.mock_project()
+        agg_cursor = _AggregatesCursor(project=project)
+        assert agg_cursor._project == project
+        assert agg_cursor._filter is None
+        assert agg_cursor._doc_filter is None
+        # Since default aggregate is not present in project, hence no aggregator
+        # will be evaulated as equal and hence all the aggregates present in this project
+        # will be considered unique.
```

Comment:
I don't think I understand what you're trying to say here. The "default aggregate" is just an individual job, right?

Reply:
What this means is that all of the operations define a custom aggregator, and thus the aggregates cursor will have some aggregates with the same id. For example, `@aggregator()` and `@aggregator.groupsof(30)` will return the same aggregates (a single aggregate containing all 30 jobs in the project) but the `_AggregatesCursor` is not capable of distinguishing which aggregators are equivalent. Thus, there are 41 aggregates in the cursor, even though there are only 34 unique aggregate ids. I updated the comments and test to clarify this behavior in e8f3701e217df0615c8fc56e9540ca533fc0fbeb.